### PR TITLE
Deduplicating extended dimensions

### PR DESCRIPTION
**Pull request by BJenning**
Ive reorganized files into folders: Dashboards, Views, Includes and LookerML.

I then moved the dimensions that were being hidden in the sessions view, from shared_fields_common into a new include file, shared_fields_no_session.

I also removed a redeclaration of p_key from page_views.view.

### CODE REVIEW

File: Views/searches.view.lkml [hunk 1]
```diff
@@ -68,7 +69,7 @@ view: searches {
   dimension: search_terms {
     description: "The search term(s) that were queried."
     type:  string
-    sql:  replace(substring(${raw_search_terms}, 3, length(${raw_search_terms})-4),'+',' ') ;;
+    sql:  substring(${raw_search_terms}, 3, length(${raw_search_terms})-4) ;;
```
Comment (mark-walle) [inline Views/searches.view.lkml]: This looks like a change unrelated to deduplicating extended dimensions. Is this for work in a different ticket?

File: Views/page_views.view.lkml [hunk 2]
```diff
@@ -20,11 +21,11 @@ view: page_views {
 
   # Addititonal Drills
   dimension: browser_language {
-    drill_fields: [page_display_url]
+    drill_fields: [shared_fields_no_session.page_display_url]
```
Comment (mark-walle) [inline Views/page_views.view.lkml]: @BJenning drills on `page_display_url` do not work when scoped like this. I think all you need to do is remove the "`shared_fields_no_session.`" scope prefixing `page_display_url`. You may need to do so on all the dimensions using `page_display_url` as a drill fields.

Please test this in an explore like https://analytics.gov.bc.ca/explore/snowplow_web_block/page_views?qid=uCCZkdMnkEIV2rJRV2goqu&toggle=fil before committing.

File: Views/sessions.view.lkml [hunk 3]
```diff
@@ -181,8 +163,10 @@ view: sessions {
     label: "First Page Display URL"
     # when editing also see clicks.truncated_target_url_nopar_case_insensitive
     description: "Cleaned URL of the page where a session began without query string or standard file names like index.html"
-    sql: ${TABLE}.first_page_display_url ;;
-    group_label: "First Page"
+    sql: CASE WHEN ${TABLE}.first_page_urlpath = '/gov/search' THEN 'https://www2.gov.bc.ca/gov/search?' || ${TABLE}.first_page_urlquery
+          ELSE ${first_page_urlscheme} || '://' || ${first_page_urlhost} || regexp_replace(${first_page_urlpath}, 'index.(html|htm|aspx|php|cgi|shtml|shtm)$','')
+         END;;
+    group_label: "Page"
```
Comment (mark-walle) [inline Views/sessions.view.lkml]: This appears to be unrelated.

File: Views/sessions.view.lkml [hunk 4]
```diff
@@ -200,7 +184,7 @@ view: sessions {
   dimension: first_page_section {
     description: "The identifier for a section of a website. The part of the URL after the domain before the next '/'"
     type: string
-    sql: ${TABLE}.first_page_section ;;
+    sql: SPLIT_PART(${TABLE}.first_page_urlpath,'/',2) ;;
```
Comment (mark-walle) [inline Views/sessions.view.lkml]: This appears to be unrelated.

File: Views/sessions.view.lkml [hunk 5]
```diff
@@ -233,6 +217,17 @@ view: sessions {
...
+    description: "Temporary fix for bug in CLD-2823"
+    type: string
+    sql: CASE
+       WHEN LOWER(${TABLE}.first_page_urlhost) = 'www2.gov.bc.ca' THEN 'https://www2.gov.bc.ca'
+       WHEN LOWER(${TABLE}.first_page_urlhost) = 'intranet.gov.bc.ca' THEN 'https://intranet.gov.bc.ca'
+       ELSE LOWER(${TABLE}.first_page_urlhost) END
+    ;;
+    group_label: "Page"
+    hidden: yes
+  }
```
Comment (mark-walle) [inline Views/sessions.view.lkml]: This appears to be unrelated.

Comment (BJenning) [reply Views/sessions.view.lkml]: Removed.

File: Views/sessions.view.lkml (hunk 4 above)
Comment (BJenning) [reply Views/sessions.view.lkml]: Ive removed it.

File: Views/sessions.view.lkml (hunk 3 above)
Comment (BJenning) [reply Views/sessions.view.lkml]: Ive removed it.

File: Views/page_views.view.lkml (hunk 2 above)
Comment (BJenning) [reply Views/page_views.view.lkml]: Ive removed the filename. Its now appearing in drill down options.

File: Views/searches.view.lkml (hunk 1 above)
Comment (BJenning) [reply Views/searches.view.lkml]: Ive reverted the sql.

File: Includes/shared_fields_common.view.lkml [hunk 6]
```diff
@@ -73,30 +73,15 @@ view: shared_fields_common {
...
-  }
 
 # Location
 
   dimension: geo_country {
     type: string
     sql: ${TABLE}.geo_country ;;
     group_label: "Location"
-    drill_fields: [page_display_url]
+    drill_fields: [shared_fields_no_session.page_display_url]
```
Comment (mark-walle) [inline Includes/shared_fields_common.view.lkml]: This doesn't work right. There is no Drill Down on the explore: https://analytics.gov.bc.ca/explore/snowplow_web_block/clicks?qid=ArrBDLrZEraDdtwYfGSb77&toggle=fil

In a test branch I tried replacing this with:
```
drill_fields: [page_display_url]
```
That causes a Looker Error, but when while it doesn't let me commit with errors, I can still test the outcomes in an explore, and the explore linked to above works correctly when I do (it shows the drilldown expected).

We don't want to commit code with syntax errors (even though technically it is working, and through administrative privilege we _can_ commit error-containing code)


Instead, much like you do in `sessions.view`, I think we will have to override this `geo_country` dimension in clicks, page views, and searches to contain:
```
drill_fields: [page_display_url]
```
and then remove the line
```
     drill_fields: [shared_fields_no_session.page_display_url]
```
from this `shared_fields_common` file.

Slightly less DRY, but better than committing code that Looker's syntax tree doesn't parse without throwing an error.
